test/functional/path/add_pathsep_spec.lua

```diff
@@ -0,0 +1,22 @@
+local helpers = require('test.functional.helpers')
+local execute, eq, clear, eval, feed, ok =
+  helpers.execute, helpers.eq, helpers.clear, helpers.eval,
+  helpers.feed, helpers.ok
+
+describe('add_pathsep function', function()
+  local test_dir = 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'
+  before_each(function()
+    clear()
+    lfs.mkdir(test_dir)
+  end)
+  after_each(function()
+    lfs.rmdir(test_dir)
+  end)
+
+  it("add_pathsep()", function()
+    execute('cd ' .. test_dir)
+    execute("echo fnamemodify('', ':p')")
+    feed('<cr>')
+  end)
```

Comment:
By convention, we might make this test `it('works', function() ...`. Doesn't look like it's asserting anything, though.
```lua
eq(test_dir, eval("echo fnamemodify('', ':p')"))
```
Though, still `add_pathsep()` isn't being directly tested. It would be better to add a test under `test/unit/...`.